executor/runtime/docker/docker.go

```diff
@@ -981,12 +981,17 @@ func (r *DockerRuntime) Prepare(ctx context.Context) (err error) { // nolint: go
 	}
 	r.volumeContainers = volumeContainers
 
+	pod, podLock := r.c.Pod()
+	podLock.Unlock()
```

Comment:
I realize you just copy/pasted this, but I think there's a bug and this should have a `defer`.

Reply:
Heh, yea. this broke the build.

Reply:
What if we had a new function called `PodRO()` that returned a readonly copy of the pod, so we didn't have to worry about a nil or something, and then we could get the pod spec safely without having to worry about locks? (only if we don't need to mutate anything)

Reply:
Yeah, as long as it doesn't matter that the view is maybe inconsistent. But IIRC it doesn't since there's only the one writer. That seems much better.